### PR TITLE
[Paywalls V2] Adds `PaywallComponentsData`

### PR DESCRIPTION
**Pull request by JayShortway**
This PR adds: 
- `PaywallComponentsData`
- `ComponentsConfig`
- `PaywallComponentsConfig`
- `Background`
- `LocalizationData` with a custom serializer
- tests

### CODE REVIEW

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/Localization.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+private object LocalizationDataSerializer : KSerializer<LocalizationData> {
+    // Documentation says to use either PrimitiveSerialDescriptor or buildClassSerialDescriptor. However, we need a
+    // polymorphic descriptor that's either a primitive (string) or a class. So neither of those options fit the bill.
+    @OptIn(InternalSerializationApi::class, ExperimentalSerializationApi::class)
+    override val descriptor: SerialDescriptor = buildSerialDescriptor(
+        serialName = "LocalizationData",
+        kind = PolymorphicKind.SEALED,
+    )
```
Comment (JayShortway) [inline purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/Localization.kt:44]: I get the feeling that kotlinx.serialization doesn't quite "agree" with what we're doing, which is the same JSON key sometimes pointing to a primitive (string) and sometimes to a class (`ThemeImageUrls`). That's why I had to add these annotations to make it work.

Comment (joshdholtz) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/Localization.kt:44]: Ahhhh yeaaah, I can see this not working well in some places 😬 It felt surpringly well in Swift but 🤔 I can't see us adding more than just text or images though. Do you think this is still okay to go this way?

**Review by joshdholtz — APPROVED** [review]
:shipit:

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/Localization.kt (hunk 1 above)
Comment (JayShortway) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/Localization.kt:44]: As discussed, we'll explore changing the schema for this. I'm going to merge this now and will update the SDK separately.

**Pull request merged**